osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,189 @@
+  }
+  std::string last_run_string = assist_data.substr(120, 16);
+
+  // If timestamp is zero dont convert to UNIX Time
+  if (last_run_string == "0000000000000000") {
+    return std::string();
+  } else {
+    // swap endianess
+    std::reverse(last_run_string.begin(), last_run_string.end());
+
+    for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+      char temp = last_run_string[i];
+      last_run_string[i] = last_run_string[i + 1];
+      last_run_string[i + 1] = temp;
+    }
+
+    // Convert Windows FILETIME to UNIX Time
+    // unsigned long long last_run = std::stoull(last_run_string.c_str(), 0,
+    // 16);
+    unsigned long long last_run =
+        tryTo<unsigned long long>(last_run_string, 16).takeOr(0ull);
+    if (last_run == 0ull) {
+      LOG(WARNING) << "Failed to convert FILETIME to UNIX time.";
+      return std::string();
+    }
+    last_run = (last_run / 10000000) - 11644473600;
+
+    std::time_t last_run_time = last_run;
+
+    struct tm tm;
+    gmtime_s(&tm, &last_run_time);
+
+    auto time_str = platformAsctime(&tm);
+    return time_str;
+  }
+}
+
+// Get execution count
+int execution_num(std::string& assist_data) {
+  if (assist_data.length() <= 16) {
+    LOG(WARNING) << "Userassist execution count format is incorrect";
+    return -1;
+  }
+
+  std::string execution_count = assist_data.substr(8, 8);
+
+  // swap endianess
+  std::reverse(execution_count.begin(), execution_count.end());
+
+  for (std::size_t i = 0; i < execution_count.length(); i += 2) {
+    char temp = execution_count[i];
+    execution_count[i] = execution_count[i + 1];
+    execution_count[i + 1] = temp;
+  }
+
+  auto count = tryTo<std::size_t>(execution_count, 16);
+  if (count.isError()) {
+    LOG(WARNING) << "Error getting execution count: " << count.takeError();
+    return -1;
+  }
+  return count.get();
+}
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
+
+  for (const auto& uKey : users) {
+    auto keyType = uKey.find("type");
+    auto keyPath = uKey.find("path");
+    if (keyType == uKey.end() || keyPath == uKey.end()) {
+      continue;
+    }
+
+    std::string fullPath = keyPath->second + kFullRegPath;
+    QueryData user_assist_results;
+    queryKey(fullPath, user_assist_results);
+
+    for (const auto& rKey : user_assist_results) {
+      auto keyType = rKey.find("type");
+      auto keyPath = rKey.find("path");
+      if (keyType == rKey.end() || keyPath == rKey.end()) {
+        continue;
+      }
+
+      std::string full_path_key = keyPath->second + "\\Count";
+
+      QueryData assist_results;
+      queryKey(full_path_key, assist_results);
+
+      for (const auto& aKey : assist_results) {
+        std::string subkey = aKey.at("path");
+
+        // split reg path by \Count\ to get Key values
+        std::size_t count_key = subkey.find("Count\\");
+        std::string value_key = subkey.substr(count_key);
```

Comment:
nit: you declare `std::string` and `std::size_t` and others quite a bit, could these be `auto`?

Reply:
switched to auto